src/features/collectables/TransferCompleteScreen.tsx

```diff
@@ -0,0 +1,204 @@
+import React, { useCallback, useEffect, useMemo } from 'react'
+import { RouteProp, useNavigation, useRoute } from '@react-navigation/native'
+import { LogBox } from 'react-native'
+import Animated, { FadeIn, FadeOut } from 'react-native-reanimated'
+import { Edge } from 'react-native-safe-area-context'
+import 'text-encoding-polyfill'
+import { useTranslation } from 'react-i18next'
+import { useSelector } from 'react-redux'
+import IndeterminateProgressBar from '../../components/IndeterminateProgressBar'
+import {
+  CollectableNavigationProp,
+  CollectableStackParamList,
+} from './collectablesTypes'
+import { DelayedFadeIn } from '../../components/FadeInOut'
+import globalStyles from '../../theme/globalStyles'
+import Box from '../../components/Box'
+import ImageBox from '../../components/ImageBox'
+import ButtonPressable from '../../components/ButtonPressable'
+import Text from '../../components/Text'
+import { ww } from '../../utils/layout'
+import BackScreen from '../../components/BackScreen'
+import { useSpacing } from '../../theme/themeHooks'
+import useCollectables from '../../utils/useCollectables'
+import { RootState } from '../../store/rootReducer'
+
+LogBox.ignoreLogs([
+  'Non-serializable values were found in the navigation state',
+])
+
+type Route = RouteProp<CollectableStackParamList, 'TransferCompleteScreen'>
+
+const TransferCollectableScreen = () => {
+  const route = useRoute<Route>()
+  const navigation = useNavigation<CollectableNavigationProp>()
+  const COLLECTABLE_HEIGHT = ww
+  const backEdges = useMemo(() => ['top'] as Edge[], [])
+
+  const { t } = useTranslation()
+  const { collectable } = route.params
+  const { refresh } = useCollectables()
+  const solanaPayment = useSelector(
+    (reduxState: RootState) => reduxState.solana.payment,
+  )
+  const spacing = useSpacing()
+
+  const backgroundImageUri = useMemo(() => {
+    return collectable?.json?.image
+  }, [collectable.json])
+
+  const onReturn = useCallback(() => {
+    navigation.popToTop()
+  }, [navigation])
+
+  useEffect(() => {
+    return () => {
+      // Refresh collectables on return
+      refresh()
```

Comment:
I've never seen someone do this before. What's the intent here? Refresh when the screen disappears?

Reply:
If so, we have this hook...
```ts
    useDisappear(() => {

    })

Reply:
Yes the intent is to refresh the screen when the component unmounts but actually not needed anymore since I am triggering a refresh when the transaction succeeds. Will remove

Reply:
Done